bokehjs/src/coffee/node_main.coffee

```diff
@@ -0,0 +1,43 @@
+path = require "path"
+assert = require "assert"
+rootRequire = require("root-require")
+
+root = rootRequire.packpath.parent()
+pkg = rootRequire("./package.json")
+
+module.constructor.prototype.require = (modulePath) ->
+  assert(modulePath, 'missing path')
+  assert(typeof modulePath == 'string', 'path must be a string')
+
+  load = (modulePath) =>
+    this.constructor._load(modulePath, this)
+
+  overridePath = pkg.browser[modulePath]
+
+  if overridePath?
+    modulePath = path.join(root, overridePath)
+
+  return load(modulePath)
+
+if not (global.window? and global.document?)
+  jsdom = require('jsdom').jsdom
```

Comment:
@rgbkrk, to me, being able to repackage bokehjs with webpack and run bokehjs in node.js, are two different targets. There are two reasons bokehjs doesn't work out of the box in node.js. One are module aliases (to `src/vendor`), defined in `package.json`. This is something webpack should easily deal with on its own. The other is lack of appropriate environment in node.js, as bokehjs is a web browser library primarily. This shouldn't concern webpack at all. However, 
if webpack requires a web browser library to be a node.js library as well, then what we can do, is to make bokehjs at least import without requiring a well set up environment. I presume this is how, e.g., jquery works in webpack. 

Also, trying to repackage bokehjs in a custom way may not be as trivial as just including it in a thridparty webpack build, because some of bokehjs' functionality, like custom models, requires specific javascript bundle's encoding and additional code to be present at runtime. Note that at this point we don't use stock browserify to build bokehjs (we use a custom module labeler), and it's likely we will be using a custom approach to unify bokehjs' and custom model build, and allow for building custom, per plot, bokehjs bundles. However, if you just want to include core bokehjs' functionality, that should be easy though.